src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -253,6 +254,122 @@ ucs_status_t uct_rc_mlx5_ep_am_short(uct_ep_h tl_ep, uint8_t id, uint64_t hdr,
     return UCS_OK;
 }
 
+#if HAVE_IBV_EXP_DM
+/* DM memory should be written by 8 bytes (int64) to eliminate
+ * processor cache issues. To make this used uct_rc_mlx5_dm_copy_data_t
+ * datatype where first hdr_len bytes are filled by message header
+ * and tail is filled by head of message. */
+static void UCS_F_ALWAYS_INLINE
+uct_rc_mlx5_ep_copy_to_dm(uct_rc_mlx5_dm_copy_data_t *cache, size_t hdr_len,
+                          const void *payload, size_t length, void *dm)
+{
+    size_t len          = sizeof(*cache) - hdr_len;
+    const uint64_t *src = payload + len;
+    uint64_t *dst       = dm;
+
+    ucs_assert(sizeof(*cache) >= hdr_len);
+
+    /* copy head of payload to tail of cache */
+    memcpy(cache->in + hdr_len, payload, ucs_min(len, length));
+
+    UCS_STATIC_ASSERT(sizeof(*cache) == sizeof(cache->out));
+    UCS_STATIC_ASSERT(sizeof(cache->in) == sizeof(cache->out));
+
+    /* atomically by 8 bytes copy data to DM */
+    *(dst++) = cache->out[0];
+    *(dst++) = cache->out[1];
+
+    for (; len < length; len += sizeof(*dst)) {
+        *(dst++) = *(src++);
+    }
+}
+
+static ucs_status_t UCS_F_ALWAYS_INLINE
+uct_rc_mlx5_ep_short_dm(uct_rc_mlx5_ep_t *ep, uct_rc_mlx5_dm_copy_data_t *cache,
+                        size_t hdr_len, const void *payload, unsigned length, int stat)
```

Comment:
instead of 'int stat', need to pass the index of the stat counter which should be updated. it could be UCT_EP_STAT_AM, UCT_TL_EP_STAT_PUT, or UCT_TL_EP_STAT_EAGER (need to add EAGER because it does not exist today)
And then this function will mimic UCT_TL_EP_STAT_OP : it will increment that given counter, and also increment UCT_EP_STAT_BYTES_SHORT